From c3d31aec4e6be220395f7927daddf0fe55cdf9df Mon Sep 17 00:00:00 2001
From: Kai.Jiang <Kai.Jiang@freescale.com>
Date: Wed, 12 Jan 2011 14:56:40 +0800
Subject: [PATCH 5/5] P1021MDS: Add tdm dts binding description file.

Add description file about tdm dts binding field in Documents.

Signed-off-by: Kai.Jiang <Kai.Jiang@freescale.com>

[Extracted from the P1021MDS_20110118_TDM_Patch.iso vendor drop.
Fixed context to apply to Wind River kernel tree.]

Integrated-by: Jiang Lu <lu.jiang@windriver.com>
---
 .../powerpc/dts-bindings/fsl/cpm_qe/qe/tdm.txt     |  135 ++++++++++++++++++++
 1 files changed, 135 insertions(+), 0 deletions(-)
 create mode 100644 Documentation/powerpc/dts-bindings/fsl/cpm_qe/qe/tdm.txt

diff --git a/Documentation/powerpc/dts-bindings/fsl/cpm_qe/qe/tdm.txt b/Documentation/powerpc/dts-bindings/fsl/cpm_qe/qe/tdm.txt
new file mode 100644
index 0000000..e4ca379
--- /dev/null
+++ b/Documentation/powerpc/dts-bindings/fsl/cpm_qe/qe/tdm.txt
@@ -0,0 +1,135 @@
+* TDM (Time Division Multiplex)
+
+(I) Required properties:
+
+-#address-cells,#size-cells,#interrupt-cells : Please follow the description
+in booting-without-of.txt.
+
+tdmphy:
+-ranges : shold be tdmphy address range.
+-compatible : should be "simple-bus", "fsl,pq-mds-t1".
+
+ds26528:
+-compatible : "dallas,ds26528".
+-reg : should be ds26528 reg address offset and size on pq-mds-t1 card.
+-line-rate : should be "t1" or "e1".
+-trans-mode : should be "normal" for tdm phy normal operation;
+	      "framer-loopback" for tdm phy framer loopback operation.
+
+pld:
+-compatible :  should be "fsl,pq-mds-t1-pld".
+-reg : should be pq-mds-t1 card pld reg offset and size.
+
+Example:
+This node is under localbus node.
+
+	tdmphy@f8020000 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		#interrupt-cells = <2>;
+		device_type = "pq-mds-t1";
+		reg = <3 0 0x10000>;
+		ranges = <0 3 0 0x10000>;
+		compatible = "fsl,pq-mds-t1";
+
+		ds26528: tdm-phy@000 {
+			compatible = "dallas,ds26528";
+			reg = <0 0x2000>;
+			line-rate = "t1";
+			trans-mode = "normal";
+		};
+
+		pld: pld-reg@2000 {
+			compatible = "fsl,pq-mds-t1-pld";
+			reg = <0x2000 0x1000>;
+		};
+	}
+
+
+(II) Required properties:
+These nodes in under qe dts node.
+
+si1:
+-compatible : should be fsl,qe-si".
+-device-id : could be <1> or <2>. If qe has two si reg, the first one is <1>,
+             and the second is <2>.
+-reg : should be si reg offset and size.
+
+siram1:
+-compatible : should be "fsl,qe-siram".
+-device-id : could be <1> or <2>. If qe has two siram, the first one is <1>,
+             and the second is <2>.
+-reg : should be si ram offset and size.
+
+
+Example:
+		si1: si@700 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "fsl,qe-si";
+			device-id = <1>;
+			reg = <0x700 0x80>;
+		};
+
+		siram1: siram@1000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			compatible = "fsl,qe-siram";
+			device-id = <1>;
+			reg = <0x1000 0x800>;
+		};
+
+
+
+(III) Required properties:
+
+device_type : should be "tdm".
+compatible : should be "fsl,ucc-tdm".
+cell-index : could be 1~8. It is the ucc num.
+reg : should be offset and size of ucc.
+interrupts : should be ucc interrupt.
+rx-clock-name : take ucc.txt as reference.
+tx-clock-name : take ucc.txt as reference.
+rx-sync-clock : could be external clock "clk1" through "clk24";
+		or internal clock "brg1" through "brg16";
+		or using "rsync_pin".
+tx-sync-clock : could be external clock "clk1" through "clk24";
+		or internal clock "brg1" through "brg16";
+		or using "tsync_pin".
+tx-timeslot : tx time slot mask. It is 32 bit num for e1, and 24 bit for
+	      t1. Each bit is one time slot switch respectively.
+	      1 - enable; 0 - disable. For example, to enable time slot 2,
+	      set tx-timeslot to be <0x4>.
+rx-timeslot : rx time slot mask. It is 32 bit num for e1, and 24 bit for
+	      t1. Each bit is one time slot switch respectively.
+	      1 - enable; 0 - disable. For example, to enable time slot 2,
+	      set rx-timeslot to be <0x4>.
+pio-handle : tdm pin handle.
+tdm-framer-type : could be "t1" or "e1".
+tdm-mode : could be "normal" for normal operation; or "internal-loopback" for
+	   tsa internal loop back.
+fsl,tdm-id : tdm port id. tdm portA~H is using  <0> ~ <7> respectively.
+fsl,siram-entry-id : siram entry id. could be any of <0>, <2>, <4> ... <14>.
+		but should use the different one between different ucc.
+
+
+Example:
+		tdma: ucc@2000 {
+			device_type = "tdm";
+			compatible = "fsl,ucc-tdm";
+			cell-index = <1>;
+			reg = <0x2000 0x200>;
+			interrupts = <32>;
+			interrupt-parent = <&qeic>;
+			rx-clock-name = "clk3";
+			tx-clock-name = "clk4";
+			rx-sync-clock = "rsync_pin";
+			tx-sync-clock = "tsync_pin";
+			tx-timeslot = <0x00fffffe>;
+			rx-timeslot = <0x00fffffe>;
+			pio-handle = <&pio_tdma>;
+			tdm-framer-type = "t1";
+			tdm-mode = "normal";
+			fsl,tdm-id = <0>;
+			fsl,siram-entry-id = <0>;
+		};
-- 
1.7.0.4

